src/main/java/seedu/pill/util/Parser.java

```diff
@@ -0,0 +1,365 @@
+                    throw new PillException(ExceptionMessages.TOO_MANY_ARGUMENTS);
+                }
+                new ListCommand().execute(this.items, this.storage);
+                break;
+            case "stock-check":
+                if (splitInput.length > 2) {
+                    throw new PillException(ExceptionMessages.TOO_MANY_ARGUMENTS);
+                }
+                new StockCheckCommand(argument).execute(this.items, this.storage);
+                break;
+            default:
+                throw new PillException(ExceptionMessages.INVALID_COMMAND);
+            }
+        } catch (PillException e) {
+            PillException.printException(e);
+        }
+    }
+
+    /**
+     * Parses the user input and creates an {@code AddItemCommand} object.
+     * The input is expected to contain the item name, quantity, and optional expiry date.
+     * If a valid date is found, it must be the last element in the input.
+     * Only one date and one quantity are allowed.
+     *
+     * The method loops through the input array to determine the item name, quantity, and expiry date,
+     * applying default values when necessary (e.g., quantity defaults to 1 if not specified).
+     *
+     * @param arguments A string representing the user's input for adding an item
+     * @return An {@code AddItemCommand} containing the parsed item name, quantity, and optional expiry date.
+     * @throws PillException If the input format is invalid
+     */
+    private AddItemCommand parseAddItemCommand(String arguments) throws PillException {
+        String[] splitArguments = arguments.split("\\s+");
+
+        if (splitArguments.length == 0) {
+            throw new PillException(ExceptionMessages.INVALID_ADD_COMMAND);
+        }
+
+        Integer quantityIndex = null;
+        Integer dateIndex = null;
+
+        for (int i = 0; i < splitArguments.length; i++) {
+            String currentArgument = splitArguments[i];
+
+            if (isValidDate(currentArgument)) {
+                if (dateIndex != null) {
+                    throw new PillException(ExceptionMessages.INVALID_ADD_COMMAND);
+                }
+                dateIndex = i;
+
+                if (i != splitArguments.length - 1) {
+                    throw new PillException(ExceptionMessages.INVALID_ADD_COMMAND);
+                }
+            }
+
+            if (isANumber(currentArgument)) {
+                quantityIndex = i;
+            }
+        }
+
+        String itemName;
+        String quantityStr = null;
+        String expiryDateStr = null;
+
+        if (quantityIndex != null && quantityIndex == splitArguments.length - 1) {
+            quantityStr = splitArguments[quantityIndex];
+            expiryDateStr = null;
+            itemName = buildItemName(splitArguments, 0, quantityIndex);
+        } else if (quantityIndex != null && quantityIndex == splitArguments.length - 2
+                && isValidDate(splitArguments[quantityIndex + 1])) {
```

Comment:
Maybe you can refactor the conditional statement out to avoid complicated espressions